client/components/domains/domain-registration-suggestion/index.jsx

```diff
@@ -299,17 +299,15 @@ class DomainRegistrationSuggestion extends Component {
 		return (
 			<div className={ wrapperClassName }>
 				<div className={ titleWrapperClassName }>
-					<h3 className="domain-registration-suggestion__title">
-						<div className="domain-registration-suggestion__domain-title">
-							<span aria-label={ domain }>
-								<span className="domain-registration-suggestion__domain-title-name">
-									{ this.getFormattedDomainName( name ) }
-								</span>
-								<span className="domain-registration-suggestion__domain-title-tld">{ tld }</span>
+					<div className="domain-registration-suggestion__title">
```

Comment:
I think we should fix how VoiceOver reads it, but preserve the document structure, in which headings are used for search results.

What if we moved `h3` down to replace the span that currently defines `aria-label` for the search result?

Reply:
Thanks, Wojtek. I was considering keeping `h3` there, but it doesn't seem correct to me (from the semantic point of view) - since those domain names are not really headings.

When looking at the existing classes, the styling should be preserved even without using `h3` there. The current structure also helps us avoid using `aria` props altogether, since screen reader can read the content correctly without it.

| Before | After |
|--------|--------|
| ![Markup on 2025-05-27 at 12:26:24](https://github.com/user-attachments/assets/1dc369ca-1f06-4f84-aa67-94b53babb1b9) | ![Markup on 2025-05-27 at 12:25:22](https://github.com/user-attachments/assets/23c5901e-2c7d-4fe1-b102-2f5a6e214b3c) |